test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -955,3 +1362,37 @@ func getTunedStructuredData(profile *performancev2.PerformanceProfile) *ini.File
 	Expect(err).ToNot(HaveOccurred())
 	return cfg
 }
+
+//This function removes the cpu siblings associated with a particular core from
+//the numaCoreSiblings map once they are used in reserved,isolated or offlined parameters
+func getCPUs(numaCoreSiblings map[int]map[int][]int, coreKey int, cpu *string, sep string) {
+	for key := range numaCoreSiblings {
+		for _, c := range numaCoreSiblings[key][coreKey] {
+			*cpu += sep + strconv.Itoa(c)
+			sep = ","
+			delete(numaCoreSiblings[key], c)
+		}
+	}
+}
+
+//This function gets the list of cpus as string
+//and splits them in to ranges.
+//Example if cpu string contains "0, 1, 2, 3, 4, 5, 6, 7, 8"
+//it returns "0-4", "5-8"
+func getRanges(cpuString string) string {
```

Comment:
this func is wrong:

`➜  test_offlined go test -v      
=== RUN   TestGetRanges
    mario_test.go:23: cpus: 1,2,3,6,8,10,11 -> ranges (1-3,6-11) don't match the expected ones (1-3,6,7,10-11)
    mario_test.go:23: cpus: 1,2,3 -> ranges (1-1,2-3) don't match the expected ones (1-3)
--- FAIL: TestGetRanges (0.00s)
FAIL
exit status 1
FAIL    go_ex/.idea/argo/test_offlined  0.001s`

`func TestGetRanges(t *testing.T) {
	tcs := []struct {
		testCpus       string
		expectedRanges string
	}{
		{
			testCpus:       "1,2,3,6,8,10,11",
			expectedRanges: "1-3,6,7,10-11",
		},
		{
			testCpus:       "1,2,3",
			expectedRanges: "1-3",
		},
	}

	for _, tc := range tcs {
		ranges := getRanges(tc.testCpus)
		if ranges != tc.expectedRanges {
			t.Logf("cpus: %s -> ranges (%s) don't match the expected ones (%s)", tc.testCpus, ranges, tc.expectedRanges)
			t.Fail()
		}
	}
}`

please add ut for the helpers to be sure that everything is ok.

Reply:
I've seen right now the comments:
with that range :
mario_test.go:27: cpus: 0,1,2,3,4,5,6,7,8 -> ranges (0-3,4-8) don't match the expected ones (0-4,5-8)

Reply:
@marioferh the aim of the function is not to do a perfect split of the cpus , the ranges can be irregular . Our intention is to only test if the offlined parameter can take multiple ranges. If the ranges itself doesn't need to be equal. 
Also this test doesn't work for test cases where the cpu are just "1, 2, 3" as we will skip the test.

Reply:
@marioferh Please have a look at the updated patch, I have addressed this.

Reply:
ok, now the output of:
//Example if cpu string contains "0, 1, 2, 3, 4, 5, 6, 7, 8"
//it returns "0-4", "5-8"

is correct, where are the UT?

Reply:
@marioferh with UT, you mean unit tests , ? where do you want me to add the tests ? and how do we run it.

Reply:
the best course of action is probably to extract the helpers in a new package to be put in `test/e2e/performanceprofile/functests/utils/`, and add tests here. Speaking of which, it seems we don't have tests for the utils here in general, which is not great.

Reply:
Right, helpers are enough complex to move them to utils and make its own unit test

Reply:
True. OTOH, I also have mixed feelings about increasing the scope of this PR.

Reply:
@marioferh I have updated the patch to add tests